### ISSUE TITLE
`Unknown error: Load failed` when logging in or signing up from a reverse proxy

### ISSUE THREAD

**Issue by buresdv** [hcengineering/huly-selfhost#25]
I just spun up Huly and put the frontend behind reverse proxy. I forwarded `myIP:8087` (default for the frontend) to `dash.mydomain.org`. This domain, along with all of its subdomains, have a properly-configured and functioning SSL.

Now, when I try to log in or create a new account through the domain, I get the error `Unknown error: Load failed`. This does not happen if I access Huly directly through `myIP:8087`.

![Snímek obrazovky 2024-08-02 v 21 50 12](https://github.com/user-attachments/assets/70174b6d-fc39-4fc1-a984-677593ad00f6)

No errors are logged into any of the containers, but the console does show this error:
`Fetch API cannot load http://myIP:3000/ due to access control checks.`

I have not changed anything in the Compose file, apart from changing the port of `minio` to `9003` due to another service already running on that port.

Can this be solved in a manner that would maintain security, and without having to expose the account service?

**Comment by aonnikov** [comment]
I believe the problem is that you exposed only the front service. Huly requires not only front, but several more services to be accessible - transactor, account, collaborator.

**Comment by buresdv** [comment]
Doesn't that sound like a huge security issue? Is it really necessary to expose anything but the frontend?

**Comment by aonnikov** [comment]
That probably does, but now we have an example of configuration that exposes only the frontend. You can check it out here: https://github.com/hcengineering/huly-selfhost/tree/main/nginx

**Comment by kkleber1991** [comment]
I did several tests, I even spent the whole day checking. The problem is related to ssl. If you use https:// this error will persist. I still haven't been able to solve it because as I use Cloudflare in the domain and need ssl https:// I couldn't solve it. I use Debian 12 + Docker + Dokploy, if anyone knows how to solve this I will be grateful.

**Comment by Jonaswinz** [comment]
I also use Huly everything behind a proxy manager reverse proxy (I commented the line that exposes the port of the huly nginx container in the compose.yaml file, and let my main reverse proxy just forwards the requests to the huly nginx container at the non-exposed port 80). I also encountered the issue first, but I resolved it by setting SECURE=true in the huly.conf (and restarting the containers). When not setting this, some internal requests of Huly will be on HTTP (even though the requests are actually done via HTTPS), and then you get a mixed-content error.
